sample/src/androidTest/java/com/kaspersky/kaspressample/configurator_tests/enricher_tests/enrichers/AssertionMainSectionEnricher.kt

```diff
@@ -0,0 +1,19 @@
+package com.kaspersky.kaspressample.configurator_tests.enricher_tests.enrichers
+
+import com.kaspersky.kaspressample.configurator_tests.enricher_tests.data.EnricherTestData
+import com.kaspersky.kaspresso.enricher.MainSectionEnricher
+import com.kaspersky.kaspresso.testcases.core.testcontext.TestContext
+import com.kaspersky.kaspresso.testcases.models.info.TestInfo
+
+class AssertionMainSectionEnricher : MainSectionEnricher<EnricherTestData> {
+
+    override fun afterMainSectionRun(testInfo: TestInfo, testContext: TestContext<EnricherTestData>) {
```

Comment:
It'a not a DSL approach =(
I propose the following way to express.
Dirty draft:
```
class MainSectionEnricherImpl<Data> : MainSectionEnricher<Data> {

    private var beforeAction: (TestContext<Data>.(TestInfo) -> Unit)? = null

    fun beforeMainSectionRun(testLambda: TestContext<Data>.(TestInfo) -> Unit) {
        beforeAction = testLambda
    }

    override fun beforeMainSectionRun(testInfo: TestInfo, testContext: TestContext<Data>) {
        beforeAction?.invoke(testContext, testInfo)
    }

}

class SomeUsage {

    fun test() {
        val enricher: MainSectionEnricherImpl<Unit> = MainSectionEnricherImpl()
        enricher.beforeMainSectionRun { testInfo ->
            step("some step") {
                testInfo.testName
                // bla-bla-bla
            }
        }
    }

}
```
As a user, I want to use beforeMainSectionRun in DSL-style. So, I think, we have to provide for users a slightly different class (not interface MainSectionEnricher) to offer a possibility of writing in so sweet manner =)

Reply:
Ok, we've discussed this proposal more detailed. 
We want to add DSL approach into enrichers implementations, like this:

```
class MyEnricher : BaseMainSectionEnricher<Data>() {

     override fun enrichMainSection() {
          beforeMainSectionRun { testInfo ->
              // here,  this == TestContext<Data> , so you can add steps in the following way
              step("My step in before run") { ... }
              step("My step 2 in before run") { ... }
          }

          afterMainSectionRun { testInfo -> 
              //  this == TestContext<Data> 
          }
     }

}
```